src/desktop/OpenRAP/sdks/UserSDK.ts

```diff
@@ -69,16 +68,85 @@ export class UserSDK {
     }
     user.updatedOn = Date.now();
     return this.dbSDK.updateDoc(USER_DB, user._id, user)
-    .then(data => ({_id: data.id}))
-    .catch(err => { throw this.dbSDK.handleError(err); });
+      .then(data => ({ _id: data.id }))
+      .catch(err => { throw this.dbSDK.handleError(err); });
+  }
+
+  // Logged in users
+  public async insertLoggedInUser(user: ISignedInUser) {
+    const userExist = await this.findByName(user.userId);
+    if (!_.isEmpty(userExist)) {
+      throw {
+        code: "UPDATE_CONFLICT",
+        status: 409,
+        message: `User already exist with id ${user.userId}`
+      }
+    }
+    user._id = uuid();
+    return this.dbSDK.insertDoc(USER_DB, user, user._id)
+      .catch(err => { throw this.dbSDK.handleError(err); });
+  }
+
+  public async getLoggedInUser(userId: string) {
+    const users = await this.findByUserId(userId);
+    return users[0];
+  }
+
+  public async updateLoggedInUser(user: ISignedInUser) {
+    if (_.get(user, '_id')) {
+      return this.updateDoc(user);
+    } else if (_.get(user, 'userId')) {
+      const userData = await this.findByUserId(user.userId);
+      user._id = _.get(userData, '_id');
+      return this.updateDoc(user);
+    } else {
+      throw {
+        code: "BAD_REQUEST",
+        status: 400,
+        message: `_id is mandatory to update user`
+      }
+    }
+
   }
-  private async findByName(name){
+
+  public deleteLoggedInUser(id: string) {
+    if (!id) {
+      throw {
+        code: "BAD_REQUEST",
+        status: 400,
+        message: `_id is mandatory to update user`
+      }
+    }
+    return this.dbSDK.delete(USER_DB, id);
+  }
+
+  public async getUserToken() {
+    return await this.settingSDK.get('oauth_token');
```

Comment:
This is not oauth_token it will be a session you will set and get and token should be part of user object only and should be omited while sending user object to frontend